src/reducers/profile-view.js

```diff
@@ -559,43 +557,23 @@ export const getRightClickedThreadIndex = (state: State) =>
 export const getSelection = (state: State) =>
   getProfileViewOptions(state).selection;
 
-export type SelectorsForThread = {
-  getThread: State => Thread,
-  getViewOptions: State => ThreadViewOptions,
-  getTransformStack: State => TransformStack,
-  getTransformLabels: State => string[],
-  getRangeFilteredThread: State => Thread,
-  getRangeAndTransformFilteredThread: State => Thread,
-  getJankInstances: State => TracingMarker[],
-  getProcessedMarkersThread: State => Thread,
-  getTracingMarkers: State => TracingMarker[],
-  getTracingMarkersForView: State => TracingMarker[],
-  getMarkerTiming: State => MarkerTimingRows,
-  getRangeSelectionFilteredTracingMarkers: State => TracingMarker[],
-  getRangeSelectionFilteredTracingMarkersForHeader: State => TracingMarker[],
-  getFilteredThread: State => Thread,
-  getRangeSelectionFilteredThread: State => Thread,
-  getCallNodeInfo: State => CallNodeInfo,
-  getCallNodeMaxDepth: State => number,
-  getSelectedCallNodePath: State => CallNodePath,
-  getSelectedCallNodeIndex: State => IndexIntoCallNodeTable | null,
-  getExpandedCallNodePaths: State => PathSet,
-  getExpandedCallNodeIndexes: State => Array<IndexIntoCallNodeTable | null>,
-  getCallTree: State => CallTree.CallTree,
-  getStackTimingByDepth: State => StackTiming.StackTimingByDepth,
-  getCallNodeMaxDepthForFlameGraph: State => number,
-  getFlameGraphTiming: State => FlameGraph.FlameGraphTiming,
-  getFriendlyThreadName: State => string,
-  getThreadProcessDetails: State => string,
-  getSearchFilteredMarkers: State => MarkersTable,
-  unfilteredSamplesRange: State => StartEndRange | null,
-};
```

Comment:
So I have some concerns:

1. having this defined statically made Flow show the error here if we returned the wrong thing. While now we'll have to connect the dots between the final user (the react component) and the consumer (whatever function in ProfileData).
   For example when I worked on `getTracingMarkersForView` below this was helping.
2. It made it easy to see what's being returned by a function. I assume that your editor shows you this information, mine doesn't... which isn't a very compelling argument I must admit :)

I understand this reduces the churn we have to add when adding new selectors here. But I'm not sure this is worth losing what I outlined above.
That said I'm very happy to discuss this. What I'd like to hear from you and other team members is what are the advantages you think we gain from this change besides having to add a line when adding a selector.

Reply:
Just to be clear: I think the cons outweight the pros at the moment. But please convince me :-)

Reply:
1. I brought this up specifically because the error Ola got when adding a selector was a bit confusing on her end, as there is a bit of misdirection with how the type was created.

2. Have implicit returns is bad, I agree, and this current solution is not enough. I'm throwing up another patch where they are enforced inline, although it would be more work to add than what I've done here. So it would need to take place in a separate PR before this one. Let me know your thoughts on going forward with this path.

I have a pretty strong preference to define and enforce types where they are created as it makes things more obvious, easier to understand, and easier to maintain.

Here are some errors, with the new `Selector<ReturnType>` type.

<img width="784" alt="image" src="https://user-images.githubusercontent.com/1588648/42887885-9b9445ea-8a6c-11e8-8dc6-c3bfbf18f48a.png">

<img width="789" alt="image" src="https://user-images.githubusercontent.com/1588648/42887973-cdf78146-8a6c-11e8-9650-adcd237cf8f0.png">

<img width="789" alt="image" src="https://user-images.githubusercontent.com/1588648/42888057-001f6ddc-8a6d-11e8-84be-d4427b51a662.png">

Reply:
Yeah this looks better. But I also agree this is some more work.

Maybe a middle way that could work now is to make sure that for all selectors, the last function has its return type defined. This is mostly the case today, and this wouldn't apply to functions defined in other files which is a bit bad.